src/state/selection/logics.ts

```diff
@@ -32,7 +32,7 @@ const storeColorsLogic = createLogic({
             }
             return newAgent;
         });
-        dispatch(setDefaultUIData(newUiData));
+        dispatch(setSelectedUIDisplayData(newUiData));
         done();
```

Comment:
Still a bit bummed to see this synchronous state derivation code in a logic. Everything this code does would work in a reducer branch, and would probably be more at home there, where it wouldn't have to re-dispatch a new action and immediately call `done` to clear out resources held by `redux-logic` for what it assumes to be an `async` operation. That change is well out of scope for this PR though.

Reply:
This seems like a good opportunity for a refactor. This logic was put in place during previous work on color changing, and its role has changed.

We can derive the new `UiDisplayData` and pass it into `redux` directly, redux never needs to handle the `ColorChange`. I am inclined to do this in `ModelPanel` instead of in the reducer because the pattern in this repo is for reducers to simply pass on their action payload without side effects.

Adding (synchronous) logic to a reducer feels like breaking that pattern in a way that puts code in _more places_ which is part of the pain of redux, for me at least. But I accept that this `redux-logic` is not the right place for it either.

Reply:
Wrote a brief issue (#596 ) and will address in a separate PR.